src/qtoolkit/io/shell.py

```
@@ -230,6 +234,10 @@ def parse_jobs_list_output(self, exit_code, stdout, stderr) -> list[QJob]:
 
             qjob = QJob()
             qjob.job_id = data[0]
+            # If the ps command truncates the username, a "+" will be in username
+            # Consider having the possibility to set a larger output for username (currently 32 characters)
+            if "+" in data[1]:
+                raise RuntimeError(f'The username was truncated: "{data[1]}".')
```

I wonder if raising is the right approach here. The user for example is not necessarily needed in jobflow-remote and if for any reason the username is longer than 32 you are unable to use the worker for no good reason, unless being able to modify the source code of qtk.

---

I would argue that there is no good reason to have a username longer than 32 characters :D one thing we may consider is to add a "scheduler_kwargs" in the config of jobflow-remote (of a worker) so that in case you have to modify it (for the username length, but also for anything else, e.g. the cancel command is different ...), you can. What do you think ? For the username length, I could increase it a bit more if you think it's worth ?

---

I agree that it is quite unlikely, but my question is rather if it is really necessary to raise, rather than having just a warning.

It can also stay like this and wait to see if this happens to anybody.
The idea of adding the scheduler_kwargs to the project configuration may still be a good one, altough I don't remember if I didn't do that for a good reason from the beginning.

---

I guess that if it raises, we should provide a way to change it in jobflow-remote. At the same time, I would say it would be good that it raises because it is otherwise not sure that it can identify the user. What do you think ? I think adding the scheduler_kwargs to the project configuration would indeed be a good idea. I think we discussed it and we said we would wait for a case where it may be needed. While it is still not strictly needed now (as noone has ever had a problem), maybe we can add so that the possibility is there. What do you think ?